src/commands/Twitch/twitchsubscription.ts

```diff
@@ -41,269 +25,313 @@ type Entry = NotificationsStreamsTwitchStreamer;
+		for (const guildSubscription of subscriptionsForStreamer) {
+			lines.push(
+				`${streamer.display_name} - ${channelMention(guildSubscription.channelId)} → ${this.getSubscriptionStatus(
+					guildSubscription.subscription.subscriptionType,
+					statuses
+				)}`
+			);
 		}
 
 		return lines;
 	}
 
-	private async showAll(guildSubscriptions: NotificationsStreamTwitch[], t: TFunction) {
-		// Retrieve all subscriptions for the guild
-		if (guildSubscriptions.length === 0) this.error(LanguageKeys.Commands.Twitch.TwitchSubscriptionShowEmpty);
+	private async showAll(guild: Guild, t: TFunction) {
+		const guildSubscriptions = await this.getGuildSubscriptions(guild);
 
-		// Fetch all usernames and map them by their id.
-		const ids = guildSubscriptions.map((subscriptions) => subscriptions[0]);
-		const profiles = await this.container.client.twitch.fetchUsers(ids, []);
+		// Get all the streamer IDs for this guild, put into a Set to remove duplicates
+		// (i.e. different channels, different subscription types)
+		const streamerIds = new Set([...guildSubscriptions.map((guildSubscription) => guildSubscription.subscription.streamerId)]);
+
+		const profiles = await this.container.client.twitch.fetchUsers(streamerIds, []);
 		const names = new Map<string, string>();
 		for (const profile of profiles.data) names.set(profile.id, profile.display_name);
 
 		// Print all entries for this guild.
 		const statuses = t(LanguageKeys.Commands.Twitch.TwitchSubscriptionShowStatus);
 		const lines: string[] = [];
-		for (const subscriptions of guildSubscriptions) {
-			const name = names.get(subscriptions[0]) || t(LanguageKeys.Commands.Twitch.TwitchSubscriptionShowUnknownUser);
-			for (const subscription of subscriptions[1]) {
-				lines.push(`${name} - <#${subscription.channel}> → ${statuses[subscription.status]}`);
-			}
+
+		for (const guildSubscription of guildSubscriptions) {
+			const name = names.get(guildSubscription.subscription.streamerId) ?? t(LanguageKeys.Commands.Twitch.TwitchSubscriptionShowUnknownUser);
+
+			lines.push(
+				`${name} - ${channelMention(guildSubscription.channelId)} → ${this.getSubscriptionStatus(
+					guildSubscription.subscription.subscriptionType,
+					statuses
+				)}`
+			);
 		}
 
 		return lines;
 	}
 
-	private async getSubscriptions(entityManager: EntityManager, ...guildIds: string[]): Promise<TwitchStreamSubscriptionEntity[]> {
-		return entityManager
-			.createQueryBuilder(TwitchStreamSubscriptionEntity, 'twitchsubs')
-			.where('twitchsubs.guild_ids IN (:ids)', { ids: guildIds })
-			.getMany();
+	private async removeSubscription(streamer: TwitchHelixUsersSearchResult, subscriptionType: TwitchEventSubTypes) {
+		const { twitchSubscriptions } = this.container.db;
+
+		// Get all subscriptions for the provided streamer and subscription type
+		const subscription = await twitchSubscriptions.findOne({
+			relations: ['guildSubscription'],
+			where: {
+				streamerId: streamer.id,
+				subscriptionType
+			}
+		});
```

Comment:
When we delete the guild subscription, we have `subscriptionId`, which we can use alongside a `LIMIT 1` to not do a very costly `JOIN LEFT` every time a subscription is removed.

Reply:
I've explained this many times already. If you do not understand the code then ask it in Discord instead of leaving review comments like this.

First of all, `findOne` already is `LIMIT 1`. It is literally in the name

Secondly, no we DO need to have `relations: ['guildSubscription']`. I've explained this a million times already now. This function is to ensure that once the last guild unsubs from a streamer, we send the unsub request to Twitch. So we DO need to know which guilds are subbed with a LEFT JOIN here.

Thirdly, sure I'll change it to matching on subscriptionId but that doesn't change point 2 at all.